src/python_testing/TC_EGCTestBase.py

```diff
@@ -0,0 +1,73 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing import matter_asserts
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class ElectricalGridConditionsTestBaseHelper:
+
+    async def test_ForecastConditions(self,
+                                      endpoint: int = None,
+                                      cluster: Clusters.ElectricalGridConditions = None,
+                                      forecastConditions: list = None):
+        matter_asserts.assert_list(forecastConditions, "ForecastConditions must be a list")
+        matter_asserts.assert_list_element_type(
+            forecastConditions, cluster.Structs.ElectricalGridConditionsStruct,
+            "ForecastConditions attribute must contain ElectricalGridConditionsStruct elements",
+            allow_empty=True)
+
+        asserts.assert_less_equal(len(forecastConditions),
+                                  56, "ForecastConditions list must be less than 56 entries")
```

Comment:
please add a constant name for 56 with a comment on why 56 is a special value? Is it spec or test plan or something else?